compat/mingw.h

```diff
@@ -344,11 +385,13 @@ static inline long long filetime_to_hnsec(const FILETIME *ft)
 #ifndef __MINGW64_VERSION_MAJOR
 #define off_t off64_t
 #define lseek _lseeki64
+#ifndef _MSC_VER
```

Comment:
In the interest of cherry-pick'ability, could you extract this guard into its own commit?